Automattic/wp-calypso
# Marketplace Themes: Makoney couldn't be activated after reverting the site

#### Proposed Changes

* Handles the edge case that prevented users from activating a Marketplace theme after the site was reverted. 
* Adds the confirmation modal to inform the users that the site will be transferred to atomic.

#### Testing Instructions

* Create a site and activate Makoney
* Revert it to simple: p1674614956322819/1674361885.647269-slack-C0160HSMDQV
* Go to the `/themes/<SITE-SLUG>`
* Find Makoney and click on the menu, then click on Activate
* The transference modal should pop up
* Confirming it, should transfer your site to atomic and activate Makoney

### Screenshots
![Screen Shot 2023-04-12 at 14 28 27](https://user-images.githubusercontent.com/1234758/231537082-b4366dbe-57f1-402d-ae1e-f49aeaca3265.png)
![Screen Shot 2023-04-12 at 14 18 35](https://user-images.githubusercontent.com/1234758/231537093-ffddde3e-fef1-4c19-ba05-05a6c55a636f.png)
![Screen Shot 2023-04-12 at 14 13 45](https://user-images.githubusercontent.com/1234758/231537098-387cb238-248c-4b88-a347-cb3456060b7d.png)


Related to #72444
